tableauserverclient/server/endpoint/jobs_endpoint.py

```diff
@@ -32,21 +38,20 @@ def get(self, job_id=None, req_options=None):
         return jobs, pagination_item
 
     @api(version="3.1")
-    def cancel(self, job_id):
+    def cancel(self, job_id: str):
         id_ = getattr(job_id, "id", job_id)
```

Comment:
thanks to the `getattr` trick, `cancel` currently accepts either a `str` or a `JobItem`...

Please,
* update the type hint accordingly
* replace the `getattr` trick by the explicit `isinstance` check similar to `wait_for_job`